src/components/ClockFace.js

```diff
@@ -1,37 +1,87 @@
-import React, { useState } from 'react';
-import { v4 as uuidv4 } from 'uuid';
-import TimerPanel from './TimerPanel';
+import React, { useState, useEffect } from 'react';
+import classnames from 'classnames';
+import SettingsModal from './SettingsModal';
 
 const ClockFace = () => {
-  // currentSession is the index of the allSessions array
-  const [currentSession, setCurrentSession] = useState(0);
-  // AllSession's structure: [[min, sec], [min, sec], [min, sec], [min, sec]]
-  // these will be set by the user
-  const [allSessions, setAllSessions] = useState([[0, 10], [1, 0], [0, 20], [0, 30]]);
-
-  const handleNextSession = () => {
-    if (allSessions.length - 1 === currentSession) {
-      // this is the last session, so go back to display the first session, without auto starting
-      setCurrentSession(0)
-    } else {
-      setCurrentSession(currentSession + 1)
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isActive, setIsActive] = useState(false);
+  const [currentSessionIndex, setCurrentSessionIndex] = useState(0);
+  // in seconds
+  const [sessions, setSessions] = useState([60, 5, 10, 10]);
+  const [currentSeconds, setCurrentSeconds] = useState(sessions[currentSessionIndex]);
+
+  useEffect(() => {
+    let timeInterval;
+    if (isActive) {
+      timeInterval = setInterval(() => {
+        if (currentSeconds > 0) {
+          setCurrentSeconds(currentSeconds - 1);
+        } else {
+          setIsActive(false);
+          clearInterval(timeInterval);
+        }
+      }, 1000)
+    }
+    else if (!isActive) {
+      clearInterval(timeInterval)
+    }
+    return () => clearInterval(timeInterval)
+  }, [isActive, currentSeconds]);
+
+  useEffect(() => {
+    if (isActive && currentSeconds === 0) {
+      setCurrentSessionIndex(currentSessionIndex + 1);
+    }
+    else if (currentSessionIndex === sessions.length) {
+      setCurrentSessionIndex(0);
+      setIsActive(false)
+    }
+  }, [currentSeconds, isActive])
```

Comment:
This is the last session, go back and display the first session without starting it